### PR TITLE
Feat/add dropdown component

### PR DESCRIPTION
**Pull request by andrewKapustin22**

### CODE REVIEW

**Review by aleksgrin — APPROVED** [review]

**Review by ACherkas12 — APPROVED** [review]

File: src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  top: 0,
+  right: 0,
+  bottom: 0,
+  left: 0,
+};
+
+export interface DropdownProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onKeyDown'> {
+  targetRef: React.RefObject<HTMLElement>;
+  relativeRef?: React.RefObject<HTMLElement>;
+  alignSelf?: 'auto' | 'flex-start' | 'flex-end' | 'center' | 'baseline' | 'stretch';
```
Comment (Kovallevz) [inline src/components/Dropdown/Dropdown.tsx:22]: Move to type.

File: src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      const viewportWidth = window.innerWidth;
+
+      const mayPlaceBottom =
+        targetRect.bottom + reltiveRect.top + DROPDOWN_OFFSET + containerRect.height + BOTTOM_HEIGHT < viewportHeight;
+      const mayPlaceTop = DROPDOWN_OFFSET + containerRect.height + HEADER_HEIGHT < targetRect.top + reltiveRect.top;
+
+      if (mayPlaceBottom) setDisplay('bottom');
+      else if (mayPlaceTop) setDisplay('top');
+      else setDisplay('center');
+      // TODO: Добавить проверку, если элемент слишком большой и не помещается ни сверху ни снизу
```
Comment (Kovallevz) [inline src/components/Dropdown/Dropdown.tsx:49]: Translate into English.

File: src/components/Dropdown/styled.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    () => {
+      switch (display) {
+        case 'bottom':
+          return 'column';
+        case 'top':
+          return 'column-reverse';
+        default:
+          return 'column';
+      }
+    }};
```
Comment (Kovallevz) [inline src/components/Dropdown/styled.ts:21]: Make a separation of styles.

File: src/components/Dropdown/styled.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+`;
+
+export const FakeTarget = styled.div`
+  pointer-events: none;
+  height: 100%;
+  flex: 0 0 auto;
```
Comment (Kovallevz) [inline src/components/Dropdown/styled.ts:27]: Make a separation of styles.

File: src/components/Dropdown/styled.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  &::-webkit-scrollbar-thumb {
+    border-radius: 10px;
+    background-color: #c0c0c0;
+    margin: 8px;
+  }
+
+  &::-webkit-scrollbar-track {
+    border-radius: 10px;
+    background-color: transparent;
+  }
```
Comment (Kovallevz) [inline src/components/Dropdown/styled.ts:69]: Make a separation of styles.

File: src/components/DropdownOption/styled.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+  font-family: 'Roboto';
+  font-size: 20px;
+  line-height: 24px;
+  color: #454545;
+
+  word-break: break-word;
+  padding: 8px 16px;
+  background-color: transparent;
+  cursor: pointer;
```
Comment (Kovallevz) [inline src/components/DropdownOption/styled.ts:16]: Make a separation of styles.

**Review by Kovallevz — CHANGES_REQUESTED** [review]

**Pull request merged**